### PR TITLE
Update error page components to make them reusable

### LINKED ISSUE #1657: Update error page components to make them reusable
We want to make a reusable component for our error pages (e.g. https://www.c0d3.com/admin/lessons)

<details>
<summary>Right now the current page displays the following</summary>
<br>

</details>

<details>
<summary>Right now the current page displays the following</summary>
<br>

</details>

We can update [Error component](https://github.com/garageScript/c0d3-app/blob/master/components/Error.tsx) to be reusable and beautify it so it can be used where applicable.

### PR DESCRIPTION
**Pull request by HS-90**
**Title:** enhancement : Update error page components to make them reusable
**Description:** closes #1657

This PR updates the error component to be more uniform and reusable. Before error pages looked very different from one another depending on the type of error. This PR adds some styling and consistency to the displayed errors:

404 error:
image

500 error:
image

403 error:
image

**Edit 8/17/2022:**
The screenshots have been updated. Some test files needed minor changes after tracing every file that uses the Error component.

### CODE REVIEW

**Comment by anthonykhoa** [comment]
1) I don't think it's a good idea to have `curriculum, repo, learn, community` buttons on error pages because it is redundant since the navbar has all those items already.

2) Why is there a new `ErrorCard` component created, instead of updating the `Error` component?

File: components/ErrorCard.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    <Layout title={errorTitle[code]}>
+      <div className={styles.container}>
+        <Image src={`/assets/errors/${code}.svg`} height={360} width={360} />
+        <div className={styles.container_text_buttons}>
+          <h1>{code + ' Error!'}</h1>
+          <Text size="lg">{errorTitle[code]}</Text>
+
+          <div className={styles.container_buttons}>
+            <NavLink
+              path="/curriculum"
```
Comment (anthonykhoa) [inline components/ErrorCard.tsx:35]: Please use constans wherever possible
https://github.com/garageScript/c0d3-app/blob/master/constants/index.ts

**Review by songz — APPROVED** [review]

File: scss/errorCard.module.scss [hunk 2]
```diff
@@ -0,0 +1,41 @@
+@use './colors.module';
```
Comment (SlyBouhafs) [inline scss/errorCard.module.scss:1]: Is is necessary to import the colors here? I don't really see it used in the file.

File: .eslintrc.json [hunk 3]
```diff
@@ -40,4 +40,4 @@
       "version": "detect"
     }
   }
-}
+}
```
Comment (anthonykhoa) [inline .eslintrc.json:43]: Can we remove this change?

File: components/Error.tsx [hunk 4]
```diff
@@ -3,47 +3,57 @@ import Layout from './Layout'
...
 export enum StatusCode {
   NOT_FOUND = 404,
+  FORBIDDEN = 403,
   INTERNAL_SERVER_ERROR = 500
 }
+
 type ErrorProps = {
   code: StatusCode
-  message?: string
+  message?: string | undefined
```
Comment (anthonykhoa) [inline components/Error.tsx:17]: if we already use typescript `?`, the `| undefined` shouldn't be necessary

File: components/Error.tsx [hunk 5]
```diff
@@ -3,47 +3,57 @@ import Layout from './Layout'
...
-              layout="responsive"
-              width={1200}
-              height={1200}
-              objectFit="contain"
-            />
+      <div className={styles.container}>
+        <Image src={`/assets/errors/${code}.svg`} height={360} width={360} />
+        <div className={styles.container_text_buttons}>
+          <h1>{code + ' Error!'}</h1>
+          <Text size="lg">{message ? message : errorTitle[code]}</Text>
```
Comment (anthonykhoa) [inline components/Error.tsx:33]: We can simplify the logic to `message || errorTitle[code]` for the text value

**Review by flacial — APPROVED** [review]

**Pull request merged**